Can't pass callback function

The following program `sample.go` triggers a panic:
```go
package main

import "github.com/containous/yaegi/interp"

const src = `package foo
func Bar(s string, fn func(string)string) string { return fn(s) }`

func main() {
	i := interp.New(interp.Options{})

	_, err := i.Eval(src)
	if err != nil {
		panic(err)
	}

	v, err := i.Eval("foo.Bar")
	if err != nil {
		panic(err)
	}

	bar := v.Interface().(func(string, func(string)string) string)

	r := bar("Kung", addS)
	println(r)
}
func addS(s string) string {
return s + "s"
}
```
See https://play.golang.org/p/jcXKEMrXgZr

Expected result:
```
Kungs
```
See https://play.golang.org/p/yJ3AdY_tSej

Got:
```
panic: reflect.Set: value of type func(string) string is not assignable to type *interp.node

goroutine 1 [running]:
reflect.Value.assignTo(0x2ad720, 0x31d8e8, 0x13, 0x2e54ea, 0xb, 0x2d5980, 0x0, 0x2d1100, 0x2b0620, 0x80c6c0, ...)
	/usr/local/go/src/reflect/value.go:2403 +0x420
reflect.Value.Set(0x2d5980, 0x80c6d8, 0x196, 0x2ad720, 0x31d8e8, 0x13)
	/usr/local/go/src/reflect/value.go:1537 +0xc0
github.com/containous/yaegi/interp.genFunctionWrapper.func2.1(0x83ee20, 0x2, 0x2, 0x83ee00, 0x83ee20, 0x8582a0, 0x1, 0x358280)
	/tmp/gopath845216020/pkg/mod/github.com/containous/yaegi@v0.7.1/interp/run.go:512 +0x400
reflect.callReflect(0x83ee00, 0x82bf58, 0x82bf4c, 0x0)
	/usr/local/go/src/reflect/value.go:549 +0x380
reflect.makeFuncStub(0x2e386d, 0x4, 0x31d8e8, 0x1, 0x2af9a0, 0x83ee00, 0x13, 0x0, 0x0, 0x78ce, ...)
	/usr/local/go/src/reflect/asm_amd64p32.s:20 +0x40
main.main()
	/tmp/sandbox709996617/prog.go:23 +0x160

```

--- a/interp/interp_eval_test.go
+++ b/interp/interp_eval_test.go
@@ -392,7 +392,7 @@ func TestEvalFunctionCallWithFunctionParam(t *testing.T) {
 	`)
 
 	v := eval(t, i, "Bar")
-	bar := v.Interface().(func(string, func(string)string) string)
+	bar := v.Interface().(func(string, func(string) string) string)
 
 	got := bar("hello ", func(s string) string {
 		return s + "world!"

--- a/interp/interp_eval_test.go
+++ b/interp/interp_eval_test.go
@@ -385,6 +385,25 @@ func TestEvalChan(t *testing.T) {
 	})
 }
 
+func TestEvalFunctionCallWithFunctionParam(t *testing.T) {
+	i := interp.New(interp.Options{})
+	eval(t, i, `
+		func Bar(s string, fn func(string)string) string { return fn(s) }
+	`)
+
+	v := eval(t, i, "Bar")
+	bar := v.Interface().(func(string, func(string)string) string)
+
+	got := bar("hello ", func(s string) string {
+		return s + "world!"
+	})
+
+	want := "hello world!"
+	if got != want {
+		t.Errorf("unexpected result of function eval: got %q, want %q", got, want)
+	}
+}
+
 func TestEvalMissingSymbol(t *testing.T) {
 	defer func() {
 		r := recover()

--- a/interp/run.go
+++ b/interp/run.go
@@ -679,9 +679,13 @@ func genFunctionWrapper(n *node) func(*frame) reflect.Value {
 
 			// Copy function input arguments in local frame
 			for i, arg := range in {
-				if def.typ.arg[i].cat == interfaceT {
+				typ := def.typ.arg[i]
+				switch {
+				case typ.cat == interfaceT:
 					d[i].Set(reflect.ValueOf(valueInterface{value: arg.Elem()}))
-				} else {
+				case typ.cat == funcT && arg.Kind() == reflect.Func:
+					d[i].Set(reflect.ValueOf(genFunctionNode(arg)))
+				default:
 					d[i].Set(arg)
 				}
 			}
@@ -703,6 +707,10 @@ func genFunctionWrapper(n *node) func(*frame) reflect.Value {
 			return result
 		})
 	}
+}
+
+func genFunctionNode(v reflect.Value) *node {
+	return &node{kind: funcType, action: aNop, rval: v, typ: &itype{cat: valueT, rtype: v.Type()}}
 }
 
 func genInterfaceWrapper(n *node, typ reflect.Type) func(*frame) reflect.Value {

